cli/src/trusted_base_cli/commands/get_person.rs

```diff
@@ -0,0 +1,61 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use crate::{
+	trusted_cli::TrustedCli, trusted_command_utils::get_pair_from_str,
+	trusted_operation::perform_trusted_operation, Cli,
+};
+use codec::Decode;
+use ita_stf::{TrustedGetter, TrustedOperation};
+use itp_stf_primitives::types::KeyPair;
+use log::{debug, info, warn};
+use sp_core::Pair;
+
+#[derive(Parser)]
+pub struct GetPersonCommand {
+	/// AccountId in ss58check format
+	account: String,
+}
+
+impl GetPersonCommand {
+	pub(crate) fn run(&self, cli: &Cli, trusted_args: &TrustedCli) {
+		println!("Result: {:?}", get_person(cli, trusted_args, &self.account));
+	}
+}
+
+pub(crate) fn get_person(cli: &Cli, trusted_args: &TrustedCli, arg_who: &str) -> Option<String> {
+	debug!("arg_who = {:?}", arg_who);
+	let who = get_pair_from_str(trusted_args, arg_who);
+	let top: TrustedOperation = TrustedGetter::get_person(who.public().into())
+		.sign(&KeyPair::Sr25519(Box::new(who)))
+		.into();
+
+	let res = perform_trusted_operation(cli, trusted_args, &top);
+	match res {
+		Some(value) => {
+			let value = u32::decode(&mut value.as_slice()).unwrap();
+			let value_bytes = value.to_be_bytes();
+			let value_str = String::from_utf8(value_bytes.to_vec()).unwrap();
+			info!("Found sum: {:?}", value_str);
```

Comment:
The issue you are facing is because you call:

```rust
let output: String = main();

// So this is a parity_scale_codec encoded string
return Some(output.encode();
```
But here you mess up your codec:

```rust
// You want to parity_scale_codec decode that string into a `u32`,
// which is wrong, but it happens to be the u32 representation of the first few bytes of your string.
let value = u32::decode(&mut value.as_slice()).unwrap();

// then you decode the first few bytes, which result in the partial string.
let value_bytes = value.to_be_bytes();
let value_str = String::from_utf8(value_bytes.to_vec()).unwrap();
```

What I recommend:
1. Don't return a string from your main, but a `Person`.
2. Derive the `Encode` and `Decode` on your person.
3. Here in this file simply do: `Person::decode(&mut value.as_slice())`